### ISSUE TITLE
Special tags within an HTML tag causes an error

### ISSUE THREAD

**Issue by phillipskevin** [canjs/can-view-parser#42]
Using a special tag within an HTML tag causes an error:
```
error Error: / is currently not supported within a tag.
    at Object.special (https://unpkg.com/can@3.5.1/dist/global/can.all.js:14243:31)
    at Function.HTMLParser.parseAttrs (https://unpkg.com/can@3.5.1/dist/global/can.all.js:3407:29)
    at parseStartTag (https://unpkg.com/can@3.5.1/dist/global/can.all.js:3224:24)
    at HTMLParser (https://unpkg.com/can@3.5.1/dist/global/can.all.js:3291:39)
    at stache (https://unpkg.com/can@3.5.1/dist/global/can.all.js:14089:9)
    at Function.stache.from.mustacheCore.getTemplateById [as from] (https://unpkg.com/can@3.5.1/dist/global/can.all.js:14278:29)
```

For instance, this will throw an error:

```
  <div
    {{#unless ./fooBar}}
      hidden
    {{/unless}}
  >
      The foo div
  </div>
```

But this works:

```
  <div {{#unless ./fooBar}}
      hidden
    {{/unless}}
  >
      The foo div
  </div>
```

Here is a JSBin reproducing the error: http://jsbin.com/dutezeweco/1/edit?html,js,output

**Comment by Alfredo-Delgado** [comment]
I haven't had much luck writing a breaking test:

```javascript
test('white space does not hurt special tags parsing (#42)', function() {
	var tests = [
		["start", ["div", false]],
		["special", ["#unless ./fooBar"]],
		["attrStart", ["hidden"]],
		["attrEnd", ["hidden"]],
		["special", ["/unless"]],
		["end", ["div", false]],
		["chars", ["The foo div"]],
		["close",["div"]],
		["done",[]]
	];

	parser('<div {{#unless ./fooBar}}hidden{{/unless}}>The foo div</div>', makeChecks(tests));
	parser(`<div
		{{#unless ./fooBar}}
      hidden
    {{/unless}}
  >The foo div</div>`, makeChecks(tests));
	parser('<div {{#unless ./fooBar}}\n\thidden\n{{/unless}}\n>The foo div</div>', makeChecks(tests));
	parser('<div\n\t{{#unless ./fooBar}}\n\t\thidden\n\t{{/unless}}\n>The foo div</div>', makeChecks(tests));
});
```

**Comment by phillipskevin** [comment]
From looking at what happens in the JSBin, it looks like the problem is that the space gets dropped after the `<div`:

![image](https://cloud.githubusercontent.com/assets/5851984/25823893/1a38b398-3403-11e7-92c9-f4ebb09edac6.png)

**Comment by Alfredo-Delgado** [comment]
I haven't looked at this in a while, but my next attempt to address it will start here: https://github.com/canjs/can-stache/blob/master/src/mustache_core.js#L477-L513

**Comment by justinbmeyer** [comment]
@Alfredo-Delgado you should work with me on the new `can-parse` version of `can-view-parser` :-)

**Comment by phillipskevin** [comment]
We should tackle this after https://github.com/canjs/can-stache/issues/290.

**Comment by frank-dspeed** [comment]
@phillipskevin this issue could need some clean up i think maybe close and reopen as some parts are already done.

**Comment by phillipskevin** [comment]
It looks like this is fixed in 4.0. Someone can re-open if they run into the same issue.
